apps/src/lab2/views/components/PredictSummary.tsx

```diff
@@ -17,18 +21,53 @@ const PredictSummary: React.FunctionComponent = () => {
     `viewAs=${ViewType.Instructor}`
   );
   const summaryUrl = document.location.pathname + SUMMARY_PATH + params;
+  const currentSectionId = useSelector(
+    (state: {teacherSections: {selectedSectionId: number}}) =>
+      state.teacherSections.selectedSectionId
+  );
+  const currentLevelId = useAppSelector(state => getCurrentLevel(state)?.id);
+  const [responseCount, setResponseCount] = React.useState<number | null>(null);
+  const [numStudents, setNumStudents] = React.useState<number | null>(null);
+
+  useEffect(() => {
+    if (currentSectionId && currentLevelId) {
+      getSectionSummary(currentSectionId, currentLevelId).then(response => {
+        if (response.response.ok) {
```

Comment:
`HttpClient` which I think this function is using should throw if the response is not `ok`. So I think you'd need to add a catch here if you want to handle the error case (or try/catch if using async/await).

Reply:
good catch!